### PR TITLE
Add a https:parse:proxy:remoteip option to the base Cell (SYN-4564)

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: synapse/lib/cell.py [hunk 1]
```diff
@@ -884,6 +884,11 @@ class Cell(s_nexus.Pusher, s_telepath.Aware):
             'type': 'object',
             'hidecmdl': True,
         },
+        'https:parse:xheaders': {
```
Comment (mikemoritz) [inline synapse/lib/cell.py:887]: "xheaders" feels too generic to me given what this is specifically enabling, unless you were planning for all future x- headers to also be enabled by this?

Comment (vEpiphyte) [reply synapse/lib/cell.py]: renamed per discussion

File: synapse/lib/cell.py [hunk 2]
```diff
@@ -2584,7 +2589,10 @@ async def addHttpsPort(self, port, host='0.0.0.0', sslctx=None):
 
             sslctx = self.initSslCtx(certpath, pkeypath)
 
-        serv = self.wapp.listen(port, address=addr, ssl_options=sslctx)
+        kwargs = {
+            'xheaders': self.conf.reqConfValu('https:parse:proxy:remoteip')
```
Comment (invisig0th) [inline synapse/lib/cell.py:2593]: Does this option have any additional side effects we need to account for? If so, we could always add a `getPeerAddr()` method or something that parses the headers ourselves.

Comment (vEpiphyte) [reply synapse/lib/cell.py:2593]: The request protocol (which identifies if a request was made with `https` vs `http` ) is also respected if the `X-Forwarded-Proto` or `X-Scheme` headers are present.

We do not use that value in any of our handlers.

**Review by Cisphyx — APPROVED** [review]

**Review by mikemoritz — APPROVED** [review]

**Review by invisig0th — APPROVED** [review]

**Pull request merged**